packages/safe-core-sdk/src/ethereumLibs/Web3Adapter.ts

```diff
@@ -65,33 +69,35 @@ class Web3Adapter implements EthAdapter {
     }
     const safeContract = this.getContract(
       contractAddress,
-      safeSingletonDeployment?.abi as AbiItem[]
+      customContractAbi ?? (safeSingletonDeployment?.abi as AbiItem[])
```

Comment:
Do you have to cast this because of the conflicting types?

Reply:
It seems you have to do this a few times in this file?

Reply:
I did the cast because the `abi` is obtained from the `safe-deployment` library, where the object returned has an `abi` with type `any[]`:
```js
export interface SingletonDeployment {
    defaultAddress: string;
    version: string;
    abi: any[];
    networkAddresses: Record<string, string>;
    contractName: string;
    released: boolean;
}
```
To instantiate the contract I call the `getContract(address: string, abi: AbiItem[])` method, where an `AbiItem[]` is expected.